king_phisher/client/windows/compare_campaigns.py

```diff
@@ -0,0 +1,155 @@
+	def __init__(self, *args, **kwargs):
+		super(CampaignCompWindow, self).__init__(*args, **kwargs)
+		treeview = self.gobjects['treeview_campaigns']
+		self.campaigns_enabled = list()
+		tvm = managers.TreeViewManager(
+			treeview,
+			cb_refresh=self.load_campaigns
+		)
+		toggle_renderer = Gtk.CellRendererToggle()
+		toggle_renderer.connect('toggled', self.signal_renderer_toggled)
+		stack_switcher = self.gobjects['stackswitcher']
+		stack_switcher.connect('button-release-event', self.show_options)
+		self.box_stack = self.gobjects['box_stack']
+		self.stack = self.gobjects['stack_1']
+		self.signal_ready = False
+		b = Gtk.CellRendererText()
+		tvm.set_column_titles(
+			('Compare', 'Name', 'Company', 'Type', 'Created By', 'Creation Date', 'Expiration'),
+			column_offset=1,
+			renderers=(toggle_renderer, b, b, b, b, b, b)
+		)
+		self._model = Gtk.ListStore(str, bool, str, str, str, str, str, str, Gdk.RGBA, Gdk.RGBA, str)
+		self._model.set_sort_column_id(2, Gtk.SortType.DESCENDING)
+		treeview.set_model(self._model)
+		self.load_campaigns()
+		self.window.resize(750, 500)
+		self.window.show_all()
+
+
+	def load_campaigns(self):
+		"""Load campaigns from the remote server and populate the :py:class:`Gtk.TreeView`."""
+		store = self._model
+		store.clear()
+		style_context = self.window.get_style_context()
+		bg_color = gui_utilities.gtk_style_context_get_color(style_context, 'theme_color_tv_bg', default=ColorHexCode.WHITE)
+		fg_color = gui_utilities.gtk_style_context_get_color(style_context, 'theme_color_tv_fg', default=ColorHexCode.BLACK)
+		for campaign in self.application.rpc.remote_table('campaigns'):
+			company = campaign.company
+			if company:
+				company = company.name
+			created_ts = utilities.datetime_utc_to_local(campaign.created)
+			created_ts = utilities.format_datetime(created_ts)
+			campaign_type = campaign.campaign_type
+			if campaign_type:
+				campaign_type = campaign_type.name
+			expiration_ts = campaign.expiration
+			if expiration_ts is not None:
+				expiration_ts = utilities.datetime_utc_to_local(campaign.expiration)
+				expiration_ts = utilities.format_datetime(expiration_ts)
+			store.append((
+				str(campaign.id),
+				False,
+				campaign.name,
+				company,
+				campaign_type,
+				campaign.user_id,
+				created_ts,
+				expiration_ts,
+				(bg_color),
+				(fg_color),
+				(html.escape(campaign.description, quote=True) if campaign.description else None)
+			))
+
+	def signal_renderer_toggled(self, _, path):
+		name = self._model[path][2]  # pylint: disable=unsubscriptable-object
+		if self._model[path][1]:  # pylint: disable=unsubscriptable-object
+			self._model[path][1] = False  # pylint: disable=unsubscriptable-object
+			self.campaigns_enabled.remove(name)
+		else:
+			self._model[path][1] = True  # pylint: disable=unsubscriptable-object
+			self.campaigns_enabled.append(name)
+		self.signal_ready = False
+		if len(self.campaigns_enabled) > 1:
+			self.signal_ready = True
+		self.init_graph()
```

Comment:
The graph should **not** be redrawn when a campaign is toggled, but rather when the stack switches from the selection treeview to the graph.